modules/backend/models/ExportModel.php

```diff
@@ -53,16 +55,17 @@ public function export($columns, $options)
      */
     public function download($name, $outputName = null)
     {
-        if (!preg_match('/^oc[0-9a-z]*$/i', $name)) {
-            throw new ApplicationException(Lang::get('backend::lang.import_export.file_not_found_error'));
-        }
-
-        $csvPath = temp_path() . '/' . $name;
-        if (!file_exists($csvPath)) {
-            throw new ApplicationException(Lang::get('backend::lang.import_export.file_not_found_error'));
-        }
-
-        return Response::download($csvPath, $outputName)->deleteFileAfterSend(true);
+        $cachedExport = Cache::get('export::' . $name);
+        return new StreamedResponse(
+            function () use ($cachedExport) {
+                echo $cachedExport;
```

Comment:
@Samuell1 Could this just use the `Cache::get` call above? You'll need to change the `use` parameter above to use `$name`.
```suggestion
                echo Cache::get('export::' . $name);
```